pya/agen/lib.py

```diff
@@ -237,6 +237,55 @@ def _generate_single(self, sample_count, start):
         return result / (10*np.sqrt(2)) # empiric scaling to match rms of sc3 PinkNoise 
 
 
+class LFImpulse(SingleChannelGen):
+    """
+    Non-band-limited single sample impulses.
+
+    Parameters
+    ----------
+    freq
+        The frequency of the oscillator in Hz. range (0Hz..Nyquist frequency).
+    phase
+        The phase of the oscillator in cycles (0..1). When modulated downwards
+        faster than the phase increase of the freq, only zero-samples are returned.
+
+
+    LFImpulse will output a 1.0 on the first sample (if phase == 0).
+    If the initial freq is 0, a single impulse is output on first sample, 
+    followed by silence until the frequency changes.
+    """
+    def __init__(
+        self,
+        freq: GenOrNum = 440.0,
+        phase: GenOrNum = 0.0,
+        *args,
+        **kwargs,
+    ):
+        super().__init__(*args, **kwargs)
+
+        self._add_node(freq, "freq", convert_num_to_arr=True)
+        self._add_node(phase, "phase", convert_num_to_arr=True)
+
+    def _generate_single(self, sample_count: int, start: int = 0) -> np.ndarray:
+        # To ensure first output is 1 if input phase == 0
+        m_phase = self.state.data.get("m_phase", -sys.float_info.min)
+        m_input_phase = self.state.data.get("m_input_phase", self.nodes["phase"][0]) # type: ignore
+
+        # Use a cumsum here to account for varying frequencies
+        phases = np.cumsum(
+            np.concatenate(
+                ([m_phase], np.minimum(self.nodes["freq"], self.sr/2) / self.sr)
```

Comment:
as this is an LF (aka non-bandlimited) version of Impulse, I wonder whether we should forego checks for Nyquist. (which would also make computation slightly faster.
Related to the comment above (line 272): this check does not prevent freqs below -self.sr/2.